ibis/expr/types/relations.py

```diff
@@ -384,14 +399,25 @@ def aggregate(
         """
         import ibis.expr.analysis as an
 
-        metrics = util.promote_list(metrics)
-        metrics.extend(
-            self._ensure_expr(expr).name(name) for name, expr in kwargs.items()
+        metrics = itertools.chain(
```

Comment:
Can we add an `aggregation_from` or `metric_from` rule? Seems like that could be reused a couple of times. 

Fine in a follow-up too.

Reply:
Let's do it in a follow-up.